src/bootstrap/compile.rs

```diff
@@ -48,6 +48,18 @@ pub fn std(build: &Build, target: &str, compiler: &Compiler) {
     let out_dir = build.cargo_out(compiler, Mode::Libstd, target);
     build.clear_if_dirty(&out_dir, &build.compiler_path(compiler));
     let mut cargo = build.cargo(compiler, Mode::Libstd, target, "build");
+    std_cargo(build, compiler, target, &mut cargo);
+    run_cargo(build,
+              &mut cargo,
+              &libstd_stamp(build, &compiler, target));
+}
+
+/// Configures the `cargo` command with env vars and such to build the standard
+/// library.
+pub fn std_cargo(build: &Build,
```

Comment:
What do you think about making `{std,test,rustc}_cargo` call Cargo themselves? Probably more than we want to do in this patch, but I believe it'd make the code cleaner, could you add a FIXME or file an issue?